### PR TITLE
fix Data race #55

### PR DESCRIPTION
**Pull request by ghostiam**

### CODE REVIEW

File: dsl_test.go [hunk 1]
```diff
@@ -750,6 +751,23 @@ var _ = Describe("MockDisplay", func() {
 			})
 		})
 
+		Context("Data race", func(){
+			It("detect", func() {
```
Comment (petergtz) [inline dsl_test.go]: How about:

```go
Context("Concurrent access to mock", func() {
    It("does not panic", ...
```

**Review by petergtz — CHANGES_REQUESTED** [review]
Hey @GhostRussia,

thanks a lot for this contribution! I have indeed completely ignored data races. Shame on me.

Before I merge your PR, would you mind making 2 small additional changes?

To exercise your new test properly, we should pass the `-race` argument to `ginkgo` in `scripts/run_tests.sh`. This script gets executed by travis.

And then a minor renaming of `Context` and `It` (see comment below).

Thanks,
Peter

**Comment by ghostiam** [comment]
> To exercise your new test properly, we should pass the -race argument to ginkgo in scripts/run_tests.sh. This script gets executed by travis.

You already run tests with the `-race` argument, just you did not have an appropriate test

File: dsl_test.go (hunk 1 above)
Comment (ghostiam) [reply dsl_test.go]: Thanks, I'm having trouble with the name of the tests. :-)

**Comment by ghostiam** [comment]
@petergtz done.

**Review by petergtz — APPROVED** [review]

**Comment by petergtz** [comment]
> You already run tests with the -race argument, just you did not have an appropriate test

Ha! Yes. *facepalm*

Merging!

Thanks again,
Peter

**Comment by ghostiam** [comment]
Thank you

**Pull request merged**